apollo-router/src/query_planner/fetch.rs

```diff
@@ -364,13 +660,23 @@ impl FetchNode {
             parameters.supergraph_request,
             parameters.schema,
             &self.input_rewrites,
+            &self.context_rewrites,
         ) {
             Some(variables) => variables,
             None => {
                 return (Value::Object(Object::default()), Vec::new());
             }
         };
 
+        let query_batched_query = if let Ok(qbq) =
+            query_batching_for_contextual_args(operation.as_serialized(), &contextual_args)
+        {
+            qbq
+        } else {
+            // TODO: do we need to provide an error here ?
+            return (Default::default(), Default::default());
+        };
```

Comment:
I don't know if we want to do this right in the `fetch_node` implementation. Is this a good idea? Should this live a step up from here? Just feels like we are modifying an operation righttttt before it gets sent off.